sympy/solvers/solveset.py

```diff
@@ -847,6 +847,9 @@ def solveset(f, symbol=None, domain=S.Complexes):
 
     if not free_symbols:
         b = Eq(f, 0)
+        if isinstance(symbol, Number):
```

Comment:
Probably I can use the following instead in Line 848
`if not free_symbols and (getattr(symbol, 'is_Symbol', False) or symbol is None):`